include/swift/Runtime/Config.h

```diff
@@ -38,7 +38,7 @@
 #ifndef SWIFT_USE_SWIFTCALL
 // Clang doesn't support mangling functions with the swiftcall attribute
 // on Windows and crashes during compilation: http://bugs.llvm.org/show_bug.cgi?id=32000
-#if (__has_attribute(swiftcall) || defined(__linux__)) && !defined(_WIN32)
+#if __has_attribute(swiftcall) && !defined(_WIN32)
```

Comment:
Ah I see. I'm happy to do the more-proper thing, just wasn't clear on what you meant; do you think it'd be adequate to extend (say) include/swift/ABI/System.h with a bunch of new macros? I don't imagine there are _too_ many targets.

(I'm also happy to tablegen something; I haven't learned how to to do that yet, but it wouldn't surprise me if there's more than one bit of information the ABI needs to specify per-target)